google-cloud-pubsublite/src/main/java/com/google/cloud/pubsublite/internal/wire/AssignerImpl.java

```diff
@@ -41,29 +44,42 @@
 
   @VisibleForTesting
   AssignerImpl(
-      PartitionAssignmentServiceStub stub,
+      StreamFactory<PartitionAssignmentRequest, PartitionAssignment> streamFactory,
       ConnectedAssignerFactory factory,
       InitialPartitionAssignmentRequest initialRequest,
       PartitionAssignmentReceiver receiver)
-      throws StatusException {
+      throws ApiException {
     this.receiver = receiver;
     this.connection =
         new RetryingConnectionImpl<>(
-            stub::assignPartitions,
+            streamFactory,
             factory,
             PartitionAssignmentRequest.newBuilder().setInitial(initialRequest).build(),
             this);
     addServices(this.connection);
   }
 
+  public AssignerImpl(
+      PartitionAssignmentServiceClient client,
+      InitialPartitionAssignmentRequest initialRequest,
+      PartitionAssignmentReceiver receiver)
+      throws ApiException {
+    this(
+        stream -> client.assignPartitionsCallable().splitCall(stream),
+        new ConnectedAssignerImpl.Factory(),
+        initialRequest,
+        receiver);
+    addServices(backgroundResourceAsApiService(client));
+  }
+
   @Override
   public void triggerReinitialize() {
     try (CloseableMonitor.Hold h = monitor.enter()) {
       connection.reinitialize();
     }
   }
 
-  private static Set<Partition> toSet(PartitionAssignment assignment) throws StatusException {
+  private static Set<Partition> toSet(PartitionAssignment assignment) {
```

Comment:
ApiException is unchecked. I introduced CheckedApiException for internal usage because I believe this to be a mistake, but it does not need to be declared.